hooks/use-fetch-exr.ts

```diff
@@ -26,42 +25,94 @@ export interface ExchangeRate {
   errorMessage: string | null;
 }
 
-export const useFetchExr = (date: string): ExchangeRate => {
-  const searchParams = useMemo(() => {
-    const p = new URLSearchParams({
-      format: "jsondata",
-      detail: "dataonly",
-    });
-    p.set("startPeriod", date);
-    p.set("endPeriod", date);
-    return p;
-  }, [date]);
+const fetchExchangeRate = async (date: string): Promise<number> => {
+  const searchParams = new URLSearchParams({
+    format: "jsondata",
+    detail: "dataonly",
+  });
+  searchParams.set("startPeriod", date);
+  searchParams.set("endPeriod", date);
+
+  return fetch(`${apiUrl}?${searchParams.toString()}`, { cache: "no-store" })
+    .then((res) => res.json())
+    .then(
+      (data: Response) =>
+        data.dataSets[0].series["0:0:0:0:0"].observations[0][0],
+    );
+};
 
-  const { data, isFetching, error } = useQuery({
+export const useFetchExr = (
+  date: string,
+): {
+  rate: number | null;
+  isFetching: boolean;
+  errorMessage: string | null;
+} => {
+  const {
+    data: rate,
+    isFetching,
+    error,
+  } = useQuery({
     queryKey: ["USD_EUR_EXCHANGE_RATE", date],
-    queryFn: (): Promise<Response> =>
-      fetch(`${apiUrl}?${searchParams.toString()}`).then((res) => res.json()),
+    queryFn: () => fetchExchangeRate(date),
     retry: 1,
   });
 
-  let rate: number | null = null;
-  let errorMessage: string | null = null;
-
-  if (error) {
-    errorMessage = `failed fetching exchange rate for date ${date}`;
-  }
-
-  try {
-    if (data) {
-      rate = data.dataSets[0].series["0:0:0:0:0"].observations[0][0];
-    }
-  } catch {
-    errorMessage = "failed extracting rate from api response";
-  }
-
   return {
-    rate,
+    rate: rate ?? null,
     isFetching,
-    errorMessage,
+    errorMessage: error ? "failed extracting rate from api response" : null,
   };
 };
+
+interface UseExchangeRateResponse {
+  isFetching: boolean;
+  queries: {
+    [date: string]: UseQueryResult<number, unknown>;
+  };
+  values: {
+    [date: string]: number;
+  };
+}
+const DEFAULT_EXCHANGE_RATE = 0;
+export const useExchangeRates = (dates: string[]): UseExchangeRateResponse => {
+  const results = useQueries({
+    queries: dates.map((date) => ({
+      queryKey: ["USD_EUR_EXCHANGE_RATE", date],
+      queryFn: async () => {
+        const rate = await fetchExchangeRate(date);
+        // To keep track of the date in combine, it is repeated in the result
+        return { date, rate };
+      },
+    })),
+  });
+
+  return results.reduce(
```

Comment:
can we use a regular for loop? I find `reduce` extremely hard to read when manipulating objects and not arrays